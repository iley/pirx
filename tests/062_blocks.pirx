extern func main(): int {
  var x: int;
  x = 3;

  var y: int;
  y = 42;

  {
    var x: int;
    x = 1;
    printf("inside first block: x = %d, y = %d\n", x, y);
  }

  {
    var x: int;
    x = 2;
    printf("inside second block: x = %d, y = %d\n", x, y);
  }

  printf("outside: x = %d, y = %d\n", x, y);
  return 0;
}
