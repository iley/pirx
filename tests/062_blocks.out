inside first block: x = 1, y = 42
inside second block: x = 2, y = 42
outside: x = 3, y = 42
